packages/node/test/unit/edge-handler/edge-handler-template.test.ts

```diff
@@ -0,0 +1,122 @@
+import { Headers, Response, Request } from 'node-fetch';
+import {
+  getUrl,
+  respond,
+  // @ts-ignore - this is a special patch file to allow importing from the template
+} from '../../../src/edge-functions/edge-handler-template.js';
+
+describe('edge-handler-template', () => {
+  describe('getUrl()', () => {
+    test('single `x-forwarded-proto` value', async () => {
+      expect(
+        getUrl(
+          'http://127.0.0.1:51126/api/add',
+          new Headers({
+            'x-forwarded-port': '',
+            'x-forwarded-proto': 'https',
+            'x-forwarded-host': 'somewhere.com',
+          })
+        )
+      ).toBe('https://somewhere.com/api/add');
+    });
+
+    test('multiple `x-forwarded-proto` value', async () => {
+      expect(
+        getUrl(
+          'https://127.0.0.1:51126/api/add',
+          new Headers({
+            'x-forwarded-port': '',
+            'x-forwarded-proto': 'https,http',
+            'x-forwarded-host': 'somewhere.com',
+          })
+        )
+      ).toBe('https://somewhere.com/api/add');
+    });
+
+    test('keep the path as part of the URL', async () => {
+      expect(
+        getUrl(
+          'https://127.0.0.1:51126/',
+          new Headers({
+            'x-forwarded-port': '',
+            'x-forwarded-proto': 'https,http',
+            'x-forwarded-host': 'somewhere.com',
+          })
+        )
+      ).toBe('https://somewhere.com/');
+    });
+
+    test('respect `x-forwarded-proto` with no `x-forwarded-host`', async () => {
+      expect(
+        getUrl(
+          'https://127.0.0.1:51126/api/add',
+          new Headers({
+            'x-forwarded-host': '127.0.0.1',
```

Comment:
<!-- Questions comments seek clarification.  -->
**question (non-blocking):** ‏Is the intent of this test that the `x-forwarded-host` matches the url or did you mean to remove this header in this test entirely?